EnvironmentSetup.sh

```diff
@@ -1,118 +1,31 @@
-#!/usr/bin/env bash
-# Setup script which install the Space Concordia Robotics Software team's development environment.
+#!/bin/bash
+# Setup script for Space Concordia Development Environment
 
-APPEND_TO_BASH="
+echo "Updating apt lists..."
 
-#------ ROBOTICS SETTINGS ------
-# competition mode
-#export ROS_MASTER_URI=http://172.16.1.30:11311
-#export ROS_HOSTNAME=$USER
-# local mode
-export ROS_MASTER_URI=http://localhost:11311
-export ROS_HOSTNAME=localhost
+sudo apt -qq update -y
 
-source ~/Programming/robotics-prototype/robot/rospackages/devel/setup.bash
-source ~/Programming/robotics-prototype/venv/bin/activate
-source ~/Programming/robotics-prototype/robot/basestation/config/.bash_aliases
+echo "Setting up python virtual environment and installing requirements..."
 
-#------ POST ROS INSTALLATION ------
-alias eb='nano ~/.bashrc'
-alias sb='source ~/.bashrc'
-alias gs='git status'
-alias gp='git pull'
-alias cw='cd ~/Programming/robotics-prototype/robot/rospackages/'
-alias cm='cw && catkin_make'
-"
-
-APPEND_TO_BASH_ALIASES='
-ROBOTICS_WS="/home/$USER/Programming/robotics-prototype"
-BASE="$ROBOTICS_WS/robot/basestation"
-ROVER="$ROBOTICS_WS/robot/rover"
-ROSPACKAGES="$ROBOTICS_WS/robot/rospackages"
-BASH_A="~/.bash_aliases"
-NANORC="~/.nanorc"
-
-# general shortcuts
-alias ..="cd .."
-alias b="cd -"
-alias robotics="cd $ROBOTICS_WS"
-alias base="cd $BASE"
-alias rover="cd $ROVER"
-alias arm="cd $ROVER/ArmDriverUnit"
-alias wheels="cd $ROVER/MobilePlatform"
-alias rostings="cd $ROSPACKAGES"
-alias mcu="cd $ROSPACKAGES/src/mcu_control/scripts"
-alias util="cd $ROBOTICS_WS/robot/util"'
-
-FINAL_MESSAGE="
-
-#################################
-The script will now exit, you should test the installation using these steps:
-1. Open a new terminal window to apply changes
--> it should automatically start with virtual env activated and you should be able to use aliases that you can lookup in your ~/.bashrc and ~/.bash_aliases files
-2. verify ROS installation using 'roscore'
-#################################"
-
-REPO="/home/$USER/Programming/robotics-prototype"
-ROS_VERSION="melodic"
-
-# check if in proper directory
-SCRIPT_LOCATION=$REPO/EnvironmentSetup.sh
-if [ ! -f $SCRIPT_LOCATION ]
-then
-    # -e enables text editing, \e[#m sets a text colour or background colour. \e[0 ends the edit.
-    echo -e "\e[31m\e[47mYou did not setup the repo directory correctly. Refer to README\e[0m"
-    exit 1
-fi
-
-
-# install prereqs
-sudo apt update -y
-sudo apt install python3.6-venv git python3-pip -y
-
-# necessary for `ifconfig` in env.sh
-sudo apt install net-tools
-
-# Setup venv
-python3.6 -m venv venv
+python -m venv venv
 source venv/bin/activate
 
-
-# Install Requirements
 pip install -U pip
 pip install -r requirements.txt
 
-# Setup python and allow for module imports from within repo
 python setup.py develop
 
-
-# Install ROS
+echo "Installing ROS..."
 bash scripts/install_ros.sh
```

Comment:
We should add a check here for if ROS is already installed

Reply:
That's pretty easy, will add it tonight

Reply:
Sounds good. Thanks

Reply:
After looking into it: It's safe to have two ros distros on one system (only matters which setup.bash is sourced)
If the wanted version is already installed, there is no harm in running the ros install script again, it will simply go faster and skip all the package installations.

Reply:
Isn't the script now called ros_install.sh?

Reply:
Lmao nice catch. We really need to test this script in a vm lol